### PR TITLE
[chip-test] List EDN functional features and add SiVal test plan

### PR DESCRIPTION
**Pull request by vogelpi**

### CODE REVIEW

**Comment by h-filali** [comment]
Hi @vogelpi ,
This PR looks good to me. Have you considered adding another test that induces an error/alert over the EDN ALERT_TEST/ERROR_TEST registers and then observes whether the response is correct?

**Review by rswarbrick — APPROVED** [review]

File: hw/ip/edn/data/edn.hjson
Comment (andreaskurth) [reply hw/ip/edn/data/edn.hjson:85]: (shepherd mode) I'm planning to address this by rephrasing to "To this end, EDN keeps replaying a sequence of `generate` and `reseed` commands, which can optionally come with additional data, provided by firmware via the !!RESEED_CMD and !!GENERATE_CMD registers, respectively."

File: hw/top_earlgrey/data/ip/chip_edn_testplan.hjson
Comment (andreaskurth) [inline hw/top_earlgrey/data/ip/chip_edn_testplan.hjson:39]: Suggest adding "and with non-deterministic seeds (`flag0 = kMultiBitBool4True` for Instantiate and Reseed commands)" as otherwise alerts about deterministic / non-FIPS-compliant entropy may be raised

File: hw/top_earlgrey/data/ip/chip_edn_testplan.hjson
Comment (andreaskurth) [inline hw/top_earlgrey/data/ip/chip_edn_testplan.hjson:63]: Suggest adding "and with non-deterministic seeds (`flag0 = kMultiBitBool4True` for Instantiate and Reseed commands)" as otherwise alerts about deterministic / non-FIPS-compliant entropy may be raised

**Review by andreaskurth — APPROVED** [review]
LGTM with one suggestion (two instances of it). Happy to implement this if you agree, @vogelpi

File: hw/ip/edn/data/edn.hjson
Comment (andreaskurth) [reply hw/ip/edn/data/edn.hjson:100]: (shepherd mode) I'll take care of this

**Comment by andreaskurth** [comment]
> Have you considered adding another test that induces an error/alert over the EDN ALERT_TEST/ERROR_TEST registers and then observes whether the response is correct?

I think this is a good idea. Since alerts are a common feature to many comportable IPs, I think we should autogenerate a test that verifies this for all comportable IPs that have alerts.

**Review by h-filali — APPROVED** [review]
Looks good to me!

File: hw/ip/edn/data/edn.hjson
Comment (vogelpi) [reply hw/ip/edn/data/edn.hjson]: Makes sense, thanks for raising this. I've now changed this as suggested by @andreaskurth .

File: hw/top_earlgrey/data/ip/chip_edn_testplan.hjson
Comment (vogelpi) [reply hw/top_earlgrey/data/ip/chip_edn_testplan.hjson]: Agreed and fixed as above.

**Comment by vogelpi** [comment]
I've also rebased this on current master. Before we can merge this, a second committer has to authorize the change to `edn.hjson`.

**Comment by andreaskurth** [comment]
CHANGE AUTHORIZED: hw/ip/edn/data/edn.hjson

**Comment by andreaskurth** [comment]
*kokoro* failure is spurious; all other CI checks pass. Approved --> merging

**Pull request merged**
